stdlib/builtins.pyi

```diff
@@ -1032,18 +1033,23 @@ else:
 
 def copyright() -> None: ...
 def credits() -> None: ...
-def delattr(__obj: Any, __name: str) -> None: ...
+def delattr(__obj: object, __name: str) -> None: ...
 def dir(__o: object = ...) -> list[str]: ...
 @overload
 def divmod(__x: SupportsDivMod[_T_contra, _T_co], __y: _T_contra) -> _T_co: ...
 @overload
 def divmod(__x: _T_contra, __y: SupportsRDivMod[_T_contra, _T_co]) -> _T_co: ...
+
+# The `globals` argument to `eval` has to be `dict[str, Any]` rather than `dict[str, object]` due to invariance.
+# (The `globals` argument has to be a "real dict", rather than any old mapping, unlike the `locals` argument.)
 def eval(
-    __source: str | bytes | CodeType, __globals: dict[str, Any] | None = ..., __locals: Mapping[str, Any] | None = ...
+    __source: str | bytes | CodeType, __globals: dict[str, Any] | None = ..., __locals: Mapping[str, object] | None = ...
 ) -> Any: ...
+
+# Comment above regarding `eval` applies to `exec` as well
 def exec(
-    __source: str | bytes | CodeType, __globals: dict[str, Any] | None = ..., __locals: Mapping[str, Any] | None = ...
-) -> Any: ...
+    __source: str | bytes | CodeType, __globals: dict[str, Any] | None = ..., __locals: Mapping[str, object] | None = ...
+) -> object: ...
```

Comment:
According to the [documentation](https://docs.python.org/3/library/functions.html#exec), `exec()` returns `None`.